refnx/conftest.py

```diff
@@ -22,21 +22,21 @@ def data_directory(tmpdir_factory):
         returned. If the retrieval fails then None is returned.
     """
     url = "https://github.com/refnx/refnx-testdata/archive/master.zip"
-    tmpdir = tmpdir_factory.mktemp("data")
+    tmpdir = Path(tmpdir_factory.mktemp("data"))
```

Comment:
Switching to pytest's `tmp_path_factory` might be even easier: https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html#the-tmp-path-factory-fixture